backend/apps/profile_info/selectors.py

```diff
@@ -0,0 +1,31 @@
+# apps/profile_info/selectors.py
+
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from apps.users.models import BaseUser
+from apps.profile_info.models import BaseUserProfile
+
+def get_user_profile(user_id):
+
+    try:
+        # Since BaseUserProfile is a proxy model, we can query BaseUser directly
+        profile = BaseUser.objects.get(id=user_id)
+        return profile
+    except BaseUser.DoesNotExist:
```

Comment:
Avoid catching stuff and raising errors with custom text here, as then the API won't return detailed messages. Though I think these are all redundant as they do nothing more than users/selectors so could probably be deleted.